File: src/app/models/SimCard.js


// Define collection and schema for SimCard
export interface SimCard {
    iccid:
	type : string
    imsi:
	type : string
    carrier:
	type : string
    NetworkProfiles:
 	type : [{ type: Schema.Types.ObjectId, ref: 'NetworkProfile' }]
    Tenant:
	type : Schema.Types.ObjectId
    ConnectivityPlan:
	type : Schema.Types.ObjectId
    Status:
 	type : String
#
    collection: 'simCards'
}

File: iotonaspdotnet/Domain/SimCard.cs
using iotonaspdotnet.Domain.Contracts;

namespace iotonaspdotnet.Domain;

public class SimCard
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long SimcardId { get; set; }
 public virtual string Iccid { get; set; }
 public virtual string Imsi { get; set; }
 public virtual string Carrier { get; set; }
public virtual NetworkProfile NetworkProfiles { get; set; }
public virtual Tenant Tenant { get; set; }
public virtual ConnectivityPlan ConnectivityPlan { get; set; }
 public virtual SimStatus Status { get; set; }

    public static SimCard FromRequest(SimCardRequest request) {
        return new SimCard {
            Id = model.Id,
            Iccid = request.Iccid,
            Imsi = request.Imsi,
            Carrier = request.Carrier,
            Status = request.Status,
        };
}
